### ISSUE TITLE
Persistent volume for Docker?

### ISSUE THREAD

**Issue by mariushosting** [chroxify/haptic#7]
Nice project congrats! Where is the persistent volume for Docker?

**Comment by chroxify** [comment]
Hey @mariushosting, thanks! Currently, there isn't one because all the data is stored within the browser rather than on the server. This means that the Docker is essentially just running the website/frontend, while the data remains within the browser itself

**Comment by ItsNoted** [comment]
> Hey @mariushosting, thanks! Currently, there isn't one because all the data is stored within the browser rather than on the server. This means that the Docker is essentially just running the website/frontend, while the data remains within the browser itself

How does one backup their notes then?

**Comment by chroxify** [comment]
Good question! Currently it's actually missing but I'll implement an export button later today for the web app! This will then allow to export the collection as a normal folder. 

Once syncing is out you obviously will also have the option to secure it that way.

**Comment by mariushosting** [comment]
> Good question! Currently it's actually missing but I'll implement an export button later today for the web app! This will then allow to export/download the collection as a normal folder.
> 
> Once syncing is out, you obviously will also have the option to secure it that way.

Better to have a persistent volume for the notes instead of an export button. Just my 2 cent :-))

**Comment by chroxify** [comment]
I partially agree as I want to keep the local-first aspect in the front and offer optional syncing, which then will store the notes on the server in a persistent volume

I hope that makes sense. Essentially, once syncing is implemented, this won't be a problem anymore, as it will do exactly what you're looking for and additionally allow notes to be accessible from other platforms, not just the web.

I guess just the time in-between while sync is not there yet will be a little confusing. Working to get it out as fast as possible though!

**Comment by chroxify** [comment]
> Honestly, I'd much rather have persistent storage than browser cache or storage. This means if we open another browser on a different PC our notes are gone...
> 
> I'm not sure how this is a good thing for people who use multiple devices. Even if you do implement a sync feature.

Yes, currently, they would. But what you're describing is exactly how the syncing server will work. Essentially, to store the notes in persistent storage, an additional server needs to run. This server will then store all notes created in the web app (and on other devices) in persistent storage and will be responsible for syncing between devices.

The request is being work on!

**Comment by edersong** [comment]
I agree with all suggestions above. It should be nice if there is a persistent volume to keep our data and be able to open our notes on another browser normally.
Additionally, the notes should be stored as files like note.md, note2.md to be able to import from other tools and easy to manage them.

**Comment by RealFascinated** [comment]
I would much prefer the notes to be on the server and not the client.

**Comment by daydaya** [comment]
I also agree to save notes on the server. Whether it is Obsidian or logseq and other note-taking software, synchronization problems give me a lot of headaches.

**Comment by Ardakilic** [comment]
One of the reasons I use [Joplin](https://github.com/laurent22/joplin) is it can also store the notes on the server, it has a native server code in addition to cloud providers support. I'd love to check Haptic if notes are persisted on a server.

**Comment by kdavisk6** [comment]
> Yes, currently, they would. But what you're describing is exactly how the syncing server will work. Essentially, to store the notes in persistent storage, an additional server needs to run. This server will then store all notes created in the web app (and on other devices) in persistent storage and will be responsible for syncing between devices.
> 
> The request is being work on!

I'd like to try summarizing feedback to clarify, what I think the outcome here is.

I see that users want **off-browser storage** so that they can 
* enabling longer term backups
* enable synchronization between instances

In addition, I'd like to add:
* enable "at-rest" encryption
* deeper integration with kubernetes and docker volumes

To meet these needs, a server component or application is being considered.  Did I miss anything?

In addition, are you looking or open to help here?  I have some experience in this area and willing to donate time to help with the development and support of this server component.

**Comment by cratedev** [comment]
Haptic is functionally useless if I can only access it locally from the same device/browser. For example, if I make my notes in Firefox, I can't even see them in Chrome on the same device. I'm not sure what purpose this is meant to serve, as it feels more like "local only" rather than "local first".

Definitely would love to see persistent storage and a server/client type of system. I'm hoping to migrate to this from Obsidian simply to have a webapp that I can use anywhere.

**Comment by berot3** [comment]
this is one of the reasons Obsidian is my go-to-note-takeing-app and its one of their strongest arguments. The notes are YOURS. With haptic they are local, but hidden away in the browsers underground...

**Comment by MaximilianKos** [comment]
This application has a lot of potential. However, I’m currently unable to fully utilize it because I need access to my notes across multiple devices. It would be incredibly helpful to have an option to mount a persistent volume for this.

Please consider prioritizing this feature.

**Comment by Lea3D** [comment]
I'm sorry, but I don't understand the purpose of this editor. If I can't access my own files from the server and they are wiped when I clear the browser cache, it doesn't serve my need to create wiki pages for a website.

I installed it as a docker through the community apps on UNRAID 7.0.0 beta 4.

**Comment by crypt0rr** [comment]
Please add persistent storage!

**Comment by Aero-Alice** [comment]
Was thinking this project might work well as a self-hosted way to edit obsidian vaults in the web browser, but definitely seems like would have to be able to mount a docker volume to give it access

**Comment by MaximilianKos** [comment]
I started to vibe code a persistent storage for haptic. if anyone wants to continue where i left off, feel free.
https://github.com/MaximilianKos/haptic-persist

it works with a node js backend to save files locally. basic functionality works (but very buggy), also when ran with docker. the whole browser storage part has been stripped.
